### ISSUE TITLE
MarshalTextString outputs old-style message start / end '<' instead of new-style '{'

### ISSUE THREAD

**Issue by garymm** [golang/protobuf#653]
The C++ text_format library delimits messages with `{` while the go text pkg delimits them with `<`.
C++:
https://github.com/google/protobuf/blob/6f88e12f9dee4a658913b475e7680952c6ff3cac/src/google/protobuf/text_format.cc#L1613
Go:
https://github.com/golang/protobuf/blob/f05648d464991ab1aa8cf6a499122c56f0f50f2f/proto/text.go#L338

The official docs use `{` for examples:
https://developers.google.com/protocol-buffers/docs/overview

To be consistent with C++ and the official docs, please switch to `{`.

**Comment by dsnet** [comment]
As much as `{` and `}` are preferred, we tried making this change in the past and it broke too much code. Since `<` and `>` are still valid delimiters, there's not a strong precedence to take on the cost of switching this.

**Comment by garymm** [comment]
May I ask what it broke? Or if you have a link to the discussion so I can read about it.

**Comment by garymm** [comment]
Also could you change the API to make it optionally and not-by-default use the `{` delimiter?

**Comment by dsnet** [comment]
I looked briefly into this again. Almost a thousand tests fail if we make the delimiter change. What's worse is that the current implementation uses "{" for groups, but "<" for messages. If we added an option for this, this discrepancy would also be visible through the options, which is not ideal either. Furthermore, with this many breakages, we certainly cant change the defaults.

I still would love to fix this someday, but this is very low priority.
